arbeitszeit/repositories.py

```diff
@@ -192,6 +192,20 @@ def perform(self) -> int:
         """
 
 
+class CooperationResult(QueryResult[Cooperation], Protocol):
+    def with_id(self, id_: UUID) -> Self:
+        ...
+
+    def with_name_ignoring_case(self, name: str) -> Self:
+        ...
+
+    def coordinated_by_company(self, company_id: UUID) -> Self:
+        ...
+
+    def joined_with_coordinator(self) -> QueryResult[Tuple[Cooperation, Company]]:
```

Comment:
I'd like to make a suggestion, but first I need to understand what a coordinator is.  I agree that "join" there can be confusing — but it sounds as if you _do_ mean a join in both senses, no?  In which case, I was also expecting to see a new DDL table creation statement to create a mapping from coordinator UUID to coordinator company.

Reply:
The following code is currently the only occurrence of this interface method being called. I think that it reads ok-ish.

```
            self.database_gateway.get_cooperations()                           
            .with_id(request.coop_id)                                          
            .joined_with_coordinator()                                         
            .first()                                                           
  ```

I understand that the terminology can be confusing, especially to people who are unfamiliar with SQL. On the other hand I guess that people who are familiar with SQL would not be confused by this. I am probably biased towards the second group of people. Since this name makes sense to me I have trouble finding an alternative name. Please suggest a better name and I will happily incorporate it in this change set.